src/spyglass/utils/dj_mixin.py

```diff
@@ -1018,6 +1015,113 @@ def restrict_by(
 
         return ret
 
+    # ------------------------------ Check locks ------------------------------
+
+    def exec_sql_fetchall(self, query):
+        """
+        Execute the given query and fetch the results.    Parameters
+        ----------
+        query : str
+            The SQL query to execute.    Returns
+        -------
+        list of tuples
+            The results of the query.
+        """
+        results = dj.conn().query(query).fetchall()
+        return results  # Check if performance schema is enabled
+
+    @classmethod
```

Comment:
I think making this a class method causes issues with calling `self.exec_sql_fetchall()` and `self._delete_deps`  

Pratically, I didn't see the executing process popping up when I ran this function after removing the decorator. May be simplest to to do this